### PR TITLE
docs(4.4): Essential APIs

### PR DESCRIPTION
**Pull request by WidgetsBurritos**
Docs for Essential APIs

### CODE REVIEW

**Comment by WidgetsBurritos** [comment]
@bighappyface I ended up breaking this section up into multiple files because it was becoming incredibly cluttered. There is probably a bit more work that can be done for this in the future (especially in regards to the Entity API), but I didn't want perfect to get in the way of good, and figured this should be decent enough to move forward with.

File: 4-back-end-development/4.4-essential-apis-entity.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+// Use the entity manager.
+$node = \Drupal::entityTypeManager()->getStorage('node')->create(array('type' => 'article', 'title' => 'Another node'));
+
+// To save an entity.
+$entity->save();
+
+// The following will attempt to insert a new node with the ID 5, this will fail if that node already exists.
+$node->nid->value = 5;
+$node->enforceIsNew(TRUE);
+$node->save()
```
Comment (bighappyface) [inline 4-back-end-development/4.4-essential-apis-entity.md]: Missing semicolon

File: 4-back-end-development/4.4-essential-apis-form.md [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+  return $form;
+}
+?>
+```
+
+## Altering an Existing From
+
+Drupal 8 uses the hook method similar to Drupal 7 to alter an existing form.
+
+For example, if you a form has the ID `some_form_id` and your module is named `mymodule`,
```
Comment (bighappyface) [inline 4-back-end-development/4.4-essential-apis-form.md]: Funny wording

File: 4-back-end-development/4.4-essential-apis-entity.md (hunk 1 above)
Comment (WidgetsBurritos) [inline 4-back-end-development/4.4-essential-apis-entity.md]: Thanks.  Got it.

File: 4-back-end-development/4.4-essential-apis-form.md (hunk 2 above)
Comment (WidgetsBurritos) [inline 4-back-end-development/4.4-essential-apis-form.md]: Fixed.

**Pull request merged**